bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/WlanThermoNanoHandler.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wlanthermo.internal;
+
+import static org.openhab.binding.wlanthermo.internal.WlanThermoBindingConstants.SYSTEM;
+import static org.openhab.binding.wlanthermo.internal.WlanThermoBindingConstants.SYSTEM_ONLINE;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.google.gson.Gson;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Authentication;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.util.DigestAuthentication;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.wlanthermo.internal.api.nano.data.Data;
+import org.openhab.binding.wlanthermo.internal.api.nano.settings.Settings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WlanThermoNanoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Schlipp - Initial contribution
+ */
+@NonNullByDefault
+public class WlanThermoNanoHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WlanThermoNanoHandler.class);
+
+    private WlanThermoNanoConfiguration config = new WlanThermoNanoConfiguration();
+    private HttpClient httpClient = new HttpClient();
+    private @Nullable ScheduledFuture<?> pollingScheduler;
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(WlanThermoBindingConstants.WLANTHERMO_THREAD_POOL);
```

Comment:
I had a lot of problems with RejectedExecutionExceptions, similar to #1078. Implemented is as suggested in that ticket and got no errors since...

Reply:
I don't  know why you got those errors. But your code doesnt look more complex than the average binding that uses the scheduler without problems. That means i suspect something else might be wrong. What i do see is in the method `checkConnection` you cancel the pollers in certain conditions. However this method is called from within those scheduled threads, meaning the call to cancle would cancel the thread its running on itself, and that might cause some problems. I would suggest  to have a look at how the flow goes and see if you can improve this and try using the provided scheduler. Or if you still have problems show some loging to see if we can find the cause.

Reply:
In order to get this in the last 2.5 release, I'd prefer to leave it as it is - running. 
Will conduct some tests later and check if this can be reverted to the common scheduler, ok?